### PR TITLE
tor-browser-bundle-bin: fix unbound variables

### LINKED ISSUE #171559: Tor Browser currently broken
### Describe the bug
On startup, Tor Browser doesn't start if some variables about Wayland aren't set. 

### Steps To Reproduce
1. Unset `MOZ_ENABLE_WAYLAND` and `WAYLAND_DISPLAY` (if set)
2. Launch Tor Browser in a terminal and see the error message.

### Expected behavior
Tor Browser should start.

### Additional context
This issue was caused by #168497. #171167 currently does not fix the problem.
```console
[user@system:~]$ tor-browser
/etc/profiles/per-user/wyndon/bin/tor-browser: line 81: MOZ_ENABLE_WAYLAND: unbound variable
```

In the meantime, an easy fix would be :
```console
[user@system:~]$ MOZ_ENABLE_WAYLAND= WAYLAND_DISPLAY= tor-browser
```

### Notify maintainers

Notifying peoples from the aforementioned PRs instead.
@Artturin @LutzCle 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.34-xanmod1, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0`
 - channels(root): `"nixos-22.05pre347261.6d8215281b2"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by Artturin**
###### Description of changes
Closes https://github.com/NixOS/nixpkgs/issues/171559

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
test with `nix run "github:artturin/nixpkgs/torbrow1#tor-browser-bundle-bin"`

**Comment by xaverdh** [comment]
I can confirm, that this fixes https://github.com/NixOS/nixpkgs/issues/171559

File: pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix [hunk 1]
```diff
@@ -409,9 +409,7 @@ stdenv.mkDerivation rec {
     LD_LIBRARY_PATH=$libPath $TBB_IN_STORE/TorBrowser/Tor/tor --version >/dev/null
 
     echo "Checking tor-browser wrapper ..."
-    DISPLAY="" MOZ_ENABLE_WAYLAND="" WAYLAND_DISPLAY="" XAUTHORITY="" \
-      XCURSOR_PATH="" XDG_RUNTIME_DIR="" XDG_SESSION_TYPE="" \
-      DBUS_SESSION_BUS_ADDRESS="" TBB_HOME=$(mktemp -d) \
+      TBB_HOME=$(mktemp -d) \
```
Comment (xaverdh) [inline pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:412]: So while this does indeed work on my system, I am wondering why DISPLAY, XAUTHORITY and DBUS_SESSION_BUS_ADDRESS were unset here prior to https://github.com/NixOS/nixpkgs/pull/168497.
Maybe this used to be there for platforms where builds are not sandboxed?

Comment (Artturin) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:412]: yeah maybe

sandbox builds were enabled on linux by default in [2019](https://nixos.org/manual/nix/stable/release-notes/rl-2.2.html?highlight=sandbox#release-22-2019-01-11) while those were added in [2017](https://github.com/nixos/nixpkgs/commit/ecd0e1a2c736a9a609cd820de0cb2861129482c3)

Comment (xaverdh) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:412]: But other platforms (mac?) don't have sandboxing yet, so we should probably not touch it yet?

Comment (Artturin) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:412]: this package only supports x86_64-linux and i686-linux

Comment (xaverdh) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:412]: I think there are valid use cases for disabling the build sandbox on linux, at least there used to be, e.g. if you are stuck with a kernel you have no control over, that does not allow for user namespace. But maybe its ok to rely on the sandbox these days.. not sure

Comment (xaverdh) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:412]: https://github.com/NixOS/nix/issues/2651 seems to be such an edge case

Comment (Artturin) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:412]: its not a big enough concern for us 

im quite sure that in nixpkgs we assume that sandboxing is on

Comment (xaverdh) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:412]: If you are sure, then lets go ahead

**Pull request merged**